packages/layout/src/ChartPanel.ts

```diff
@@ -196,8 +221,87 @@ export class ChartPanel extends Border2 implements IHighlight {
+    responsivePostUpdate(mode, domNode, element) {
+        switch (mode) {
+            case "tiny":
+                element.selectAll("div.title-text,div.icon-bar").style("visibility", "hidden");
+                element.selectAll("div.data-count")
+                    .style("visibility", "visible")
+                    .style("font-size", (this.titleIconFontSize() / 3) + "px")
+                    .style("line-height", (this.titleIconFontSize() / 3) + "px")
+                    .style("left", this.titleIconFontSize() + "px")
+                    .text(this.data().length)
+                    ;
+                element.style("transform", `translate(0px,0px) scale(1)`);
+                const iconDiv = element.selectAll("div.title-icon");
+                const _node = iconDiv.node();
+                const elementRect = element.node().getBoundingClientRect();
+                if (_node) {
+                    const rect = iconDiv.node().getBoundingClientRect();
+                    const icon_top = elementRect.height / 2;
+                    iconDiv
+                        .style("position", "absolute")
+                        .style("left", `calc(50% - ${rect.width / 2}px)`)
+                        .style("top", `${icon_top - (rect.height / 2)}px`)
+                        ;
+                    element.selectAll("div.data-count")
+                        .style("position", "absolute")
+                        .style("left", `calc(50% + ${rect.width / 2}px)`)
+                        .style("top", `${icon_top - (rect.height / 2)}px`)
+                        ;
+                }
+                break;
+            case "small":
+                element.selectAll("div.title-icon").style("position", "static");
+                element.selectAll("div.body,div.title-text,div.icon-bar").style("visibility", "visible");
+                element.selectAll("div.title-text").style("display", "block");
+                element.selectAll("div.data-count").style("display", "none");
+                const rect = element.node().getBoundingClientRect();
+                const parentRect = element.node().parentElement.getBoundingClientRect();
+                const curr_transform = element.style("transform");
+                element.style("transform", `translate(${parentRect.x - rect.x}px, ${parentRect.y - rect.y}px) ${curr_transform}`);
+                break;
+            case "regular":
+                element.selectAll("div.title-icon").style("position", "static");
+                element.selectAll("div.body,div.icon-bar").style("visibility", "visible");
+                element.selectAll("div.title-text").style("display", "block");
+                element.selectAll("div.data-count").style("display", "none");
+                break;
+        }
+    }
+
     private _prevChartDataFamily;
     update(domNode, element) {
+        const _responsiveMode = this.getResponsiveMode();
+        this.responsivePreUpdate(_responsiveMode, domNode, element);
```

Comment:
Minor: It might be cleaner to have 3 methods and move the switch statement into the update method: 
* preUpdateTiny
* preUpdateSmall
* preUpdateRegular